ceph_deploy/util/pkg_managers.py

```diff
@@ -319,6 +330,16 @@ class DNF(RPMManagerBase):
     executable = 'dnf'
     name = 'dnf'
 
+    def install(self, packages, **kw):
+        extra_install_flags = kw.pop('extra_install_flags', [])
+        if '--best' not in extra_install_flags:
```

Comment:
You are right that we are forcing it...

I'm not sure to tell you the truth.  I am making it behave like Yum and apt-get, such that installing a package that is already present will upgrade it.  DNF just says "it's already there".  So it's consistent in that regard.  Let me do one more check to make sure I am right about this on Yum...